report/report.go

```diff
@@ -88,9 +89,19 @@ func (g Generator) reportObjFromDependency(dep models.Dependency) (*models.Repor
 			return nil, errors.Wrapf(err, "unable to generate reportObject from dependency %s", dep.Name)
 		}
 	case GERRIT:
-		reportObject, err = versioncontrol.ReportObjFromGerrit(dep, g.request)
+		reportObject, err = versioncontrol.ReportObjFromGerrit(dep, g.client)
 		if err != nil {
-			return nil, errors.Wrapf(err, "unable to generate reportObject from dependency %s", dep.Name)
+			//if the error is from the slack notification, we want to attempt to notify via slack
+			if strings.Contains(err.Error(), "unable to retrieve license for") {
```

Comment:
Generally we don't want to use text from an error to check what "kind" it is, it's better to define an error package variable and then wrap it when returning up the call stack so here you can do `if errors.Cause(err) == ErrNoLicence` or something like that.